### PR TITLE
chore: install ip-masq-agent as part of overlay cns scenarios

### PR DESCRIPTION
**Pull request by pjohnst5**
Install `ip-masq-agent` in go-code (rather than scripts) and tie it to overlay cns scenarios

### CODE REVIEW

File: .pipelines/singletenancy/azure-cni-overlay/azure-cni-overlay-e2e-step-template.yaml [hunk 1]
```diff
@@ -68,14 +68,7 @@ steps:
...
       echo "Start Integration Tests on Overlay Cluster"
-      echo "deploy ip-masq-agent for overlay"
-      kubectl apply -f test/integration/manifests/ip-masq-agent/ip-masq-agent.yaml --validate=false
-      cd test/integration/manifests/ip-masq-agent/
-      kubectl apply -f config-custom.yaml
-      kubectl apply -f config-reconcile.yaml
-      cd ../../../..
-      kubectl get po -owide -A
-      CNS=$(make cns-version) DROPGZ=$(make cni-dropgz-version)
+      CNS=$(make cns-version)
```
Comment (paulyufan2) [inline .pipelines/singletenancy/azure-cni-overlay/azure-cni-overlay-e2e-step-template.yaml:71]: nit: probably we can add CNS=$(make cns-version) to line 72? CNS_VERSION=$(make cns-version)

Comment (pjohnst5) [reply .pipelines/singletenancy/azure-cni-overlay/azure-cni-overlay-e2e-step-template.yaml:71]: Yeah probably, wondering if nested `make` calls would be an issue

Comment (pjohnst5) [reply .pipelines/singletenancy/azure-cni-overlay/azure-cni-overlay-e2e-step-template.yaml:71]: maybe that's a question for another PR

File: .pipelines/cni/singletenancy/linux-cniv2-template.yaml [hunk 2]
```diff
@@ -55,40 +55,35 @@ stages:
...
+                  scriptType: "bash"
+                  addSpnToEnvironment: true
+                  inlineScript: |
+                    echo "Start Integration Tests on Swift Cluster"
+                    make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}-$(commitID)
+                    kubectl cluster-info
+                    kubectl get po -owide -A
+                    sudo -E env "PATH=$PATH" make test-integration CNS_VERSION=$(cnsVersion) CNI_DROPGZ_VERSION=$(dropgzVersion) INSTALL_CNS=true INSTALL_AZURE_VNET=true TEST_DROPGZ=${{ parameters.testDropgz }}
+                name: "swifte2e"
+                displayName: "Swift Integration"
```
Comment (jpayne3506) [inline .pipelines/cni/singletenancy/linux-cniv2-template.yaml:86]: nit: you only need to indent over once.

**Review by debecerra — APPROVED** [review]

**Review by paulyufan2 — APPROVED** [review]

File: .pipelines/singletenancy/azure-cni-overlay/azure-cni-overlay-e2e-step-template.yaml (hunk 1 above)
Comment (jpayne3506) [reply .pipelines/singletenancy/azure-cni-overlay/azure-cni-overlay-e2e-step-template.yaml:71]: Nested make commands run in serial and creates another shell until the command is completed. If it fails, the exit code is passed along the stack. The only problem with make commands is if you do not use `set -e` OR if another command replaces the exit code before the `- script: |` block is done then pipelines will not catch the failure.

File: test/internal/kubernetes/utils_create.go [hunk 3]
```diff
@@ -429,6 +472,12 @@ func setupCNSDaemonset(
 		return appsv1.DaemonSet{}, cnsDetails{}, errors.Wrap(err, "failed to parse cns daemonset")
 	}
 
+	if cnsScenarioDetails.installIPMasqAgent {
```
Comment (jpayne3506) [inline test/internal/kubernetes/utils_create.go:475]: We should add a log message about `InstallIPMasqAgent`. Either inside the function or before the call.

**Review by jpayne3506 — CHANGES_REQUESTED** [review]

File: test/internal/kubernetes/utils_create.go (hunk 3 above)
Comment (pjohnst5) [reply test/internal/kubernetes/utils_create.go:475]: Done

File: .pipelines/cni/singletenancy/linux-cniv2-template.yaml (hunk 2 above)
Comment (pjohnst5) [reply .pipelines/cni/singletenancy/linux-cniv2-template.yaml]: Done

**Review by jpayne3506 — APPROVED** [review]

**Review by debecerra — APPROVED** [review]

**Pull request merged**